tf-modules/data-persistence/elasticsearch.tf

```diff
@@ -0,0 +1,117 @@
+locals {
+  inside_vpc         = length(var.subnet_ids) > 0 ? true : false
+  deploy_inside_vpc  = var.include_elasticsearch && local.inside_vpc
+  deploy_outside_vpc = var.include_elasticsearch && local.inside_vpc == false
+  include_es_policy  = length(var.es_trusted_role_arns) > 0 ? true : false
+  es_domain_name     = "${var.prefix}-${var.elasticsearch_config.domain_name}${local.inside_vpc ? "-vpc" : ""}"
+}
+
+data "aws_iam_policy_document" "es_access_policy" {
+  statement {
+    actions = [
+      "es:*"
+    ]
+
+    principals {
+      type        = "AWS"
+      identifiers =  distinct(compact(var.es_trusted_role_arns))
+    }
+
+    resources = compact(
```

Comment:
@nemreid @yjpa7145 Looks like there wasn't a circular dependency, so I updated this based on your suggestions, which also removed some unnecessary data sources

Reply:
I don't understand what purpose the `[0]` serves here?

Reply:
@nemreid The `aws_elasticsearch_domain.es_vpc` resource is using the `count =` field to conditionally create that resource. If it should be created, the count is set to 1. If it should not be created the count is set to 0.

That's not really the intended purpose of `count`, though. The intended purpose is to let you create multiple instances of a resource. When you use the count field, you are no longer creating one resource, you're creating a list of resources (from the same resource definition). At that point, you can no longer use `aws_elasticsearch_domain.es_vpc` to refer to the resource, because there could be more than one resource created there. That's why you need to use array references to refer to a specific instance. In this case we're only creating a single instance. Hopefully that helps.